### PR TITLE
Handle array of object, string number and scientific number

### LINKED ISSUE #1: Wrong parsing on scientific notation
How to reproduce

```
#1 Copy the text below

amount:2e+06  currency:"IDR"

#2 Then run

pbpaste | ./logtojson

{
  "amount": "2e",
  "currency": "IDR"
}
```

Losing the scientific notation is quite crucial i think. So, I assume the expected output should be like this isn't it?

```
{
  "amount": "2e+06",
  "currency": "IDR"
}
```

Thanks

### PR DESCRIPTION
**Pull request by wlwtvr**
Fixes #1, #2, #3

### CODE REVIEW

File: main_test.go [hunk 1]
```diff
@@ -7,35 +7,74 @@ import (
...
 			},
+			expectedError: nil,
+		},
+		{
+			name:  "Test scientific notation number",
+			input: `name:"John" age:30 salary:2e+06 married:true`,
+			expectedOutput: map[string]interface{}{
+				"name":    "John",
+				"age":     30,
+				"salary":  2000000.0,
```
Comment (alakanaeru) [inline main_test.go]:
```suggestion
				"salary":  2e+06,
```
i think we should retain user's input as it is, wdyt?

File: main_test.go [hunk 2]
```diff
@@ -7,35 +7,74 @@ import (
...
+					"Alice",
+					"Bob",
+				},
+				"age": 25,
+			},
+			expectedError: nil,
+		},
+		{
+			name:  "Test identical field name, with object value",
+			input: `display_data:{header:"header_1" title:"title_1"} display_data:{header:"header_2"  title:"title_2"}`,
```
Comment (alakanaeru) [inline main_test.go:60]: LGTM. btw to check for _potential_ breaking when _child_ has the identical key name as someone's _parent_, wdyt if we test this input?

```
name:"Agus" display_data:{name:"Samantha"}
```

File: main_test.go (hunk 1 above)
Comment (wlwtvr) [reply main_test.go]: fair enough. updated the code

File: main_test.go (hunk 2 above)
Comment (wlwtvr) [reply main_test.go:60]: test added.

**Comment by alakanaeru** [comment]
LGTM

**Pull request merged**